src/main/java/Buddy.java

```diff
@@ -0,0 +1,158 @@
+                for (int i = 0; i < taskCount; i++) {
+                    //print each line in proper format
+                    System.out.println("     " + (i + 1) + "." + tasks[i]);
+                    //System.out.println("     " + (i + 1) + ". [" + tasks[i].getStatusIcon() + "] " + tasks[i].getDescription());
+                }
+                System.out.println("____________________________________________________________");
+            }
+            else if (input.startsWith("mark")) //mark task and print acknowledge message when user marks a task
+            {
+                int taskNumber = Integer.parseInt(input.split(" ")[1]) - 1; //get the integer in the text. minus 1 to get correct index in the "tasks" array
+                tasks[taskNumber].markAsDone();
+                System.out.println("____________________________________________________________");
+                System.out.println("It’s done. Now, get lost before I toss you out of my swamp!");
+                System.out.println("     [" + tasks[taskNumber].getStatusIcon() + "] " + tasks[taskNumber].getDescription());
+                System.out.println("____________________________________________________________");
+            }
+            else if (input.startsWith("unmark")) //unmark task and print acknowledge message when user unmarks a task
+            {
+                int taskNumber = Integer.parseInt(input.split(" ")[1]) - 1; //get the integer in the text
+                tasks[taskNumber].markAsNotDone();
+                System.out.println("____________________________________________________________");
+                System.out.println("It’s unmarked. Gonna change it again? Make up your mind, or get out of my swamp!");
+                System.out.println("     [" + tasks[taskNumber].getStatusIcon() + "] " + tasks[taskNumber].getDescription());
+                System.out.println("____________________________________________________________");
+            }
+            else if (input.startsWith("todo"))
+            {
+                String description = input.substring(5); //at index 5 of the string, the description starts
+                tasks[taskCount] = new Todo(description); //put new task todo object in task array
+                taskCount++;
+                printAddedTask(tasks[taskCount - 1], taskCount);
+            }
+            else if (input.startsWith("deadline")) {
+                String[] parts = input.substring(9).split(" /by ");
+                tasks[taskCount] = new Deadline(parts[0], parts[1]);
+                taskCount++;
+                printAddedTask(tasks[taskCount - 1], taskCount);
+            }
+            else if (input.startsWith("event"))
+            {
+                String[] parts = input.substring(6).split(" /from ");
+                String[] time = parts[1].split(" /to ");
+                tasks[taskCount] = new Event(parts[0], time[0], time[1]);
```

Comment:
I like how you have split the first section of the input which is the task and the second section which contains the deadlines into different arrays !